internal/pkg/options/validators.go

```diff
@@ -0,0 +1,33 @@
+package options
+
+import (
+	"errors"
+
+	"github.com/buzzfeed/sso/internal/pkg/sessions"
+)
+
+var (
+	ErrInvalidEmailAddress = errors.New("Invalid Email Address In Session State")
+	ErrValidationError     = errors.New("Error during validation")
+)
+
+type Validator interface {
+	Validate(*sessions.SessionState) error
+	Type() string
+}
+
+func RunValidators(validators []Validator, session *sessions.SessionState) ([]string, []error) {
```

Comment:
We can clean up this function definition by just returning errors. It doesn't look like returning the validator type as a string slice buys us much?

Reply:
The main thing it provides is logging around which validators passed (without having the validators emit log messages themselves). It felt like it might be something useful to log now that we allow the request as long as one validator is successful. 

I'm happy to remove though -- perhaps the benefit isn't worth it in this case!

Reply:
^ removed!